lib/game-engine/OneVersusOne/game_page.dart

```diff
@@ -1,41 +1,114 @@
-    return const Scaffold(
-      body: Placeholder(
-        child: Text('GamePage'),
+  Widget build(BuildContext context, WidgetRef ref) {
+    final GameState gameState = ref.watch(gameProvider);
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      ref.read(gameProvider.notifier).displayEndGameOverlay(context);
+    });
+    return Scaffold(
+      appBar: generateAppBar(
+        context,
+        type: HeaderType.back,
+        isLeavingGame: true,
+        gameController: ref.read(gameProvider.notifier),
+        content: Stack(
+          alignment: Alignment.center,
+          children: [
+            Row(
+              mainAxisAlignment: MainAxisAlignment.center,
+              children: [
+                PlayerCard(
+                  reverse: false,
+                  pseudo: gameState.me!.player.pseudo,
+                  textSize: PlayerNameType.small,
+                  imgName: "profil",
+                  cardSize: CardSize.small,
+                  isLoading: false,
+                ),
+                const VersusWidget(),
+                PlayerCard(
+                  reverse: true,
+                  pseudo: gameState.opponent != null
+                      ? gameState.opponent!.player.pseudo
+                      : "",
+                  textSize: PlayerNameType.small,
+                  imgName: "profil",
+                  cardSize: CardSize.small,
+                  isLoading: false,
+                ),
+              ]
+                  .map((e) => Padding(
+                        padding: const EdgeInsets.symmetric(horizontal: 10),
+                        child: e,
+                      ))
+                  .toList(),
+            ),
+            Positioned(
+              right: 50,
+              child: Text(
+                gameState.me!.isTurn
+                    ? AppLocalizations.of(context)!.myTurn.toUpperCase()
+                    : AppLocalizations.of(context)!.ennemysTurn.toUpperCase(),
+                style: BattleShipFont.policy(context).labelLarge!.apply(
+                      color: BSColors().white,
+                    ),
+              ),
+            )
+          ],
+        ),
+      ),
+      body: SizedBox(
+        height: MediaQuery.of(context).size.height,
+        child: Row(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: [
+            Grid(
+              isDisabled: true,
+              isMyUserGrid: true,
+              isMyTurn: gameState.me!.isTurn,
+              isPlacingBoat: false,
+              shipsPlacement: gameState.me!.grid.boatsPlacement,
+              missCells: gameState.me!.grid.missCells,
+              hitCells: gameState.me!.grid.hitCells,
+              sunkenBoats: gameState.me!.grid.sunkenBoats,
+            ),
+            const SizedBox(
+              width: 30,
+            ),
+            if (gameState.opponent != null)
+              Grid(
+                isDisabled: gameState.me!.isTurn == false,
```

Comment:
```suggestion
                isDisabled: !(gameState.me!.isTurn),
```

In fact, I don't really know if that way's better